fasthtml/js.py

```diff
@@ -12,16 +13,41 @@ def MarkdownJS(sel='.marked'):
     src = "proc_htmx('%s', e => e.innerHTML = marked.parse(e.textContent));" % sel
     return Script(marked_imp+src, type='module')
 
-def KatexMarkdownJS(sel='.marked', katex_tags='$'):
-    right_tags = '\\$' if katex_tags=='$' else '\\]'
+def KatexMarkdownJS(sel='.marked', inline_delim='$', display_delim='$$', math_envs=None):
+    math_envs = math_envs or ['equation', 'align', 'gather', 'multline']
+    env_list = ','.join(f"'{env}'" for env in math_envs)
+
     src = """
     import katex from "https://cdn.jsdelivr.net/npm/katex/dist/katex.mjs";
-    const renderMath = tex => katex.renderToString(tex, {throwOnError: false, displayMode: false});
-
+    const renderMath = (tex, displayMode) => {
+        return katex.renderToString(tex, {
+            throwOnError: false,
+            displayMode: displayMode,
+            output: 'html',
+            trust: true
+        });
+    };
+    const processLatexEnvironments = (content) => {
+        return content.replace(/\\\\begin{(\\w+)}([\\s\\S]*?)\\\\end{\\1}/g, (match, env, innerContent) => {
```

Comment:
How about we change the python string to an r-string (i.e `r"""`) so then we don't need all these doubled backticks?